src/features.jl

```diff
@@ -66,7 +66,7 @@ Base.IteratorEltype(::Type{<:FeatureCollection}) = Base.HasEltype()
 
 # read only AbstractVector
 Base.size(fc::FeatureCollection) = size(array(fc))
-Base.getindex(fc::FeatureCollection, i) = array(fc)[i]
+Base.getindex(fc::FeatureCollection, i) = Feature(array(fc)[i])
```

Comment:
This is returning a type different to the `eltype` of the array, breaking the base julia array interface, and e.g. `broadcast`.

If you want it to return a `Feature`, then you will need to fix this line too:

```julia
struct FeatureCollection{O,A} <: AbstractVector{eltype(A)}
```

The eltype has to be `<: Feature`.

Reply:
This is the reason that test is still broken... I'm not sure it makes sense.

Reply:
Yeah you're right, it's fixed here: https://github.com/visr/GeoJSONTables.jl/pull/14/commits/0eb72ae2728c28ae66cdddb4f0edabc71c5cf8f3#diff-d1e20e357675f88e1ae77acf2891272308c984b6b4775fc2932e9ede2c08f162R34.

That commit is rather big, I'd appreciate it if you had a look. Still have some more things I want to iron out but we're getting close.

Reply:
The elype is abstract now as `Feature` is parametric... This will propagate type instability. 

Having the eltype of `FeatureCollection` different to its first type parameter is also usually best avoided. 

We really need to get the actual Feature type at the point of FeatureCollection construction. But this all gets pretty messy, and is why I just avoided it and went with doing the conversion in GeoInterface.

Reply:
Hmm yeah I'll look into it a bit more. Summing a property over 16k `Feature`s is 10x slower, compared to doing it over the `JSON3.Object`s.

```julia
@time sum(f -> f.SUB_AREA, fc)
# 18.695922 seconds (557.25 k allocations: 2.395 GiB, 2.00% gc time, 0.31% compilation time)
@time sum(f -> f.properties.SUB_AREA, GJ.features(fc))
# 1.991023 seconds (490.61 k allocations: 396.247 MiB, 5.79% gc time, 5.52% compilation time)
```

Reply:
Yeah I expected something like that. Also those allocations are pretty large.

Probably manually getting the type of the first `Feature` at construction time and using that as the first parameter of `FeatureCollection` and `AbstractVector` will get you back to the same speed as the raw json.

Reply:
So does that mean its not lazy now?

Reply:
No it's still lazy. On read JSON3 constructs this semi-lazy JSON3.Object for the FeatureCollection. When we do object.feature that gives a JSON3.Array which is a view into the same lazy object. Nothing is materialized and nothing is copied.

Reply:
Ok awesome O think this is a good solution then. Thanks for fixing this

Reply:
I'm happy to merge this now if you are

Reply:
I'll merge this PR into master, but there are still a few things I want to check / tidy up before putting this on top of GeoJSON.jl. I'll try to wrap it up soon and put it in a last grab-bag PR, ok?

Reply:
I can also organise the GeoJSON PR if you like. Would be good to merge that tomorrow if possible...

Reply:
Yeah would be great if you could organise the GeoJSON PR. I'll limit my actions now mostly on things that would be breaking. I also noticed that for instance writing the large geojson mentioned above took 3 minutes. Perhaps that can be improved, but it wouldn't change the API, so I'll leave that for later.